### PR TITLE
Add Dayforce connector implementation

### PR DESCRIPTION
**Pull request by niruhan**
Send and receive json objects to Dayforce REST API

## Useful Links
https://docs.google.com/spreadsheets/d/1ZMpRMGKGzEB1MAnv8npwno46UEF8dvPEZeaROHE-Ymw/edit?usp=sharing
https://usconfigr57.dayforcehcm.com/api/ddn/swagger/index.html

## Automation tests
 - Unit tests 
   > No
 - Integration tests
   > Tests written for all POST and GET methods but not for PATCH methods since the http library used in our connectorImplBase does not support patch

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -21,3 +21,6 @@
 
 # virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
 hs_err_pid*
+
+target/
+logs/
```
Comment (arunans23) [inline .gitignore]: Add new line here

File: repository/esb-connector-ceridian-dayforce.properties [hunk 2]
```diff
@@ -0,0 +1,2 @@
+apiUrl=https://usconfigr57.dayforcehcm.com/Api/ddn
+apiVersion=V1
```
Comment (arunans23) [inline repository/esb-connector-ceridian-dayforce.properties]: Add a new line here

File: src/main/assembly/assemble-connector.xml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (arunans23) [inline src/main/assembly/assemble-connector.xml]: Change the licence header.

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,389 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (Vathsan) [inline pom.xml]: Shall we update the year?

File: src/main/java/org/wso2/carbon/esb/connector/CeridianDayforceConnector.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+ * Sample method implementation.
+ */
+public class CeridianDayforceConnector extends AbstractConnector {
+
+    @Override
+    public void connect(MessageContext messageContext) throws ConnectException {
+        Object templateParam = getParameter(messageContext, "generated_param");
+        try {
+            log.info("CeridianDayforce sample connector received message :" + templateParam);
+            /**Add your connector code here
```
Comment (arunans23) [inline src/main/java/org/wso2/carbon/esb/connector/CeridianDayforceConnector.java]: Do we need this Sample code?

File: src/main/java/org/wso2/carbon/esb/mediator/CeridianDayforceMediator.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+ */
+package org.wso2.carbon.esb.mediator;
+
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.mediators.AbstractMediator;
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.ConnectException;
+
+/**
+ * Sample method implementation.
```
Comment (arunans23) [inline src/main/java/org/wso2/carbon/esb/mediator/CeridianDayforceMediator.java]: Correct the javadoc here

File: src/main/java/org/wso2/carbon/esb/mediator/CeridianDayforceMediator.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    private String clientNameSpaceKey;
+    private String serviceUri;
+
+    @Override
+    public boolean mediate(MessageContext synCtx) {
+        try {
+            uriPath = synCtx.getProperty("uri.var.uriPath").toString();
+            clientNameSpaceKey = synCtx.getProperty("uri.var.clientNamespace").toString();
+            uriPath = uriPath + clientNameSpaceKey;
+            serviceUri = synCtx.getProperty("SERVICE_URI").toString() + clientNameSpaceKey.substring(clientNameSpaceKey.lastIndexOf("/"));
```
Comment (arunans23) [inline src/main/java/org/wso2/carbon/esb/mediator/CeridianDayforceMediator.java]: Break the line here

File: src/test/java/org/wso2/carbon/connector/integration/test/CeridianDayforce/CeridianDayforceConnectorIntegrationTest.java [hunk 8]
```diff
@@ -0,0 +1,1095 @@
...
+        Assert.assertEquals(eiRestResponse.getHttpStatusCode(), 200);
+        Assert.assertEquals(apiRestResponse.getHttpStatusCode(), 200);
+        Assert.assertEquals(eiRestResponse.getBody().getJSONObject("Data").length(), apiRestResponse.getBody()
+                .getJSONObject("Data").length());
+    }
+
+    @Test(enabled = true, groups = {"wso2.ei"}, description = "GET Employee Details Test Case with contextDate")
+    public void GETEmployeeDetailsTestWithcontextDate() throws Exception {
+        RestResponse<JSONObject> eiRestResponse =
+                sendJsonRestRequest(getProxyServiceURLHttp("getEmployeeDetails") + "?xRefCode=42199&contextDate=2017-01-01T13:24:56", "GET",
```
Comment (arunans23) [inline src/test/java/org/wso2/carbon/connector/integration/test/CeridianDayforce/CeridianDayforceConnectorIntegrationTest.java]: Line length exceeded

File: src/test/java/org/wso2/carbon/connector/integration/test/CeridianDayforce/CeridianDayforceConnectorIntegrationTest.java [hunk 9]
```diff
@@ -0,0 +1,1095 @@
...
+        Assert.assertEquals(eiRestResponse.getHttpStatusCode(), 200);
+        Assert.assertEquals(apiRestResponse.getHttpStatusCode(), 200);
+        Assert.assertEquals(eiRestResponse.getBody().getJSONObject("Data").length(), apiRestResponse.getBody()
+                .getJSONObject("Data").length());
+    }
+
+    @Test(enabled = true, groups = {"wso2.ei"}, description = "POST Employee Test Case")
+    public void PostEmployeeTest() throws Exception {
+        RestResponse<JSONObject> eiRestResponse = sendJsonRestRequest(getProxyServiceURLHttp("postEmployee") +
+                        "?isValidateOnly=true", "POST", eiRequestHeadersMap, "postEmployeeRequest.json");
```
Comment (arunans23) [inline src/test/java/org/wso2/carbon/connector/integration/test/CeridianDayforce/CeridianDayforceConnectorIntegrationTest.java]: Indentation issue

File: src/test/java/org/wso2/carbon/connector/integration/test/CeridianDayforce/CeridianDayforceConnectorIntegrationTest.java [hunk 10]
```diff
@@ -0,0 +1,1095 @@
...
+
+        Assert.assertEquals(eiRestResponse.getHttpStatusCode(), 200);
+        Assert.assertEquals(apiRestResponse.getHttpStatusCode(), 200);
+        Assert.assertEquals(eiRestResponse.getBody().getString("Data"), (apiRestResponse.getBody().getString("Data")));
+    }
+
+    @Test(enabled = true, groups = {"wso2.ei"}, description = "POST Employee Addresses Test Case")
+    public void PostEmployeeAddressesTest() throws Exception {
+        RestResponse<JSONObject> eiRestResponse = sendJsonRestRequest(getProxyServiceURLHttp("postEmployeeAddresses") +
+                "?isValidateOnly=true&xRefCode=42199", "POST", eiRequestHeadersMap, "postEmployeeAddressesRequest.json");
```
Comment (arunans23) [inline src/test/java/org/wso2/carbon/connector/integration/test/CeridianDayforce/CeridianDayforceConnectorIntegrationTest.java]: Line limit exceeded

File: src/test/java/org/wso2/carbon/connector/integration/test/CeridianDayforce/CeridianDayforceConnectorIntegrationTest.java [hunk 11]
```diff
@@ -0,0 +1,1095 @@
...
+    }
+
+    @Test(enabled = true, groups = {"wso2.ei"}, description = "getEmployeeEmergencyContacts Test Case")
+    public void getEmployeeEmergencyContactsTest() throws Exception {
+        RestResponse<JSONObject> eiRestResponse =
+                sendJsonRestRequest(getProxyServiceURLHttp("getEmployeeEmergencyContacts"), "POST",
+                        eiRequestHeadersMap, "getEmployeeEmergencyContacts.json");
+
+        String apiEndPoint = connectorProperties.getProperty("apiUrl") + "/" + connectorProperties
+                .getProperty("apiVersion") + "/Employees/42199/EmergencyContacts?contextDateRangeTo=2017-01-01T13:24:56";
```
Comment (arunans23) [inline src/test/java/org/wso2/carbon/connector/integration/test/CeridianDayforce/CeridianDayforceConnectorIntegrationTest.java]: Line limit exceeded

File: src/test/java/org/wso2/carbon/connector/integration/test/CeridianDayforce/CeridianDayforceConnectorIntegrationTest.java [hunk 12]
```diff
@@ -0,0 +1,1095 @@
...
+        Assert.assertEquals(apiRestResponse.getHttpStatusCode(), 200);
+        Assert.assertEquals(eiRestResponse.getBody().getString("Data"), (apiRestResponse.getBody().getString("Data")));
+    }
+
+    @Test(enabled = true, groups = {"wso2.ei"}, description = "postEmployeeEmergencyContacts Test Case")
+    public void postEmployeeEmergencyContacts() throws Exception {
+        RestResponse<JSONObject> eiRestResponse =
+                sendJsonRestRequest(getProxyServiceURLHttp("postEmployeeEmergencyContacts"), "POST",
+                        eiRequestHeadersMap, "postEmployeeEmergencyContacts.json");
+
```
Comment (arunans23) [inline src/test/java/org/wso2/carbon/connector/integration/test/CeridianDayforce/CeridianDayforceConnectorIntegrationTest.java]: Remove empty whitespaces

File: src/test/java/org/wso2/carbon/connector/integration/test/CeridianDayforce/CeridianDayforceConnectorIntegrationTest.java [hunk 13]
```diff
@@ -0,0 +1,1095 @@
...
+        Assert.assertEquals(apiRestResponse.getHttpStatusCode(), 200);
+        Assert.assertEquals(eiRestResponse.getBody().getString("Data"), (apiRestResponse.getBody().getString("Data")));
+    }
+
+    @Test(enabled = true, groups = {"wso2.ei"}, description = "postEmployeeMaritalStatuses Test Case")
+    public void postEmployeeMaritalStatuses() throws Exception {
+        RestResponse<JSONObject> eiRestResponse =
+                sendJsonRestRequest(getProxyServiceURLHttp("postEmployeeMaritalStatuses"), "POST",
+                        eiRequestHeadersMap, "postEmployeeMaritalStatuses.json");
+
```
Comment (arunans23) [inline src/test/java/org/wso2/carbon/connector/integration/test/CeridianDayforce/CeridianDayforceConnectorIntegrationTest.java]: Remove empty white spaces

File: src/test/java/org/wso2/carbon/connector/integration/test/CeridianDayforce/CeridianDayforceConnectorIntegrationTest.java [hunk 14]
```diff
@@ -0,0 +1,1095 @@
...
+        RestResponse<JSONObject> eiRestResponse =
+                sendJsonRestRequest(getProxyServiceURLHttp("getEmployeeCompensationSummary"), "POST",
+                        eiRequestHeadersMap, "getEmployeeCompensationSummary.json");
+
+        String apiEndPoint = connectorProperties.getProperty("apiUrl") + "/" + connectorProperties
+                .getProperty("apiVersion") + "/Employees/42199/CompensationSummary";
+        RestResponse<JSONObject> apiRestResponse = sendJsonRestRequestHTTPS(apiEndPoint, "GET", apiRequestHeadersMap,
+                null, null);
+
+        Assert.assertEquals(eiRestResponse.getHttpStatusCode(), 200);
```
Comment (arunans23) [inline src/test/java/org/wso2/carbon/connector/integration/test/CeridianDayforce/CeridianDayforceConnectorIntegrationTest.java]: These three lines have been duplicated several times. Can't we do something about it?

**Pull request merged**